bgc_data/nrs_depth_binned_ctd_map.sql

```diff
@@ -26,37 +26,25 @@ ctd_profiles AS (
                                                               (nt.sampledatelocal + INTERVAL '1' DAY)
          AND dp.site_code = nt.site_code
 ),
---select the minimum absolute difference in time for every trip_code
-ctd_selection AS (
-      SELECT
-         cp.trip_code,
-         MIN(cp.absolute_time_difference) AS minimum_absolute_time_difference
-      FROM ctd_profiles cp
-      GROUP BY cp.trip_code
-),
---identify the ctd files id to join on the measurements table
-identify_files_id AS (
-      SELECT
-         cp.file_id,
-         cp.cast_time,
-         cp.trip_code
-      FROM ctd_selection cs INNER JOIN ctd_profiles cp
-            ON cs.trip_code = cp.trip_code
-            AND cs.minimum_absolute_time_difference = cp.absolute_time_difference
+--for each trip, pick the cast that is closest in time
+matched_profiles AS (
+      SELECT DISTINCT ON (trip_code) *
+      FROM ctd_profiles
+      ORDER BY trip_code, absolute_time_difference
```

Comment:
This will pick out the row with the smallest `absolute_time_difference` for each `trip_code`.
(See https://www.postgresql.org/docs/current/sql-select.html#SQL-DISTINCT)